File: container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        size = len(height)
        stack = []
        maxWater = 0
        
        left = 0
        right = size - 1
        
        while(left<right):
            area = min(height[left],height[right])*(right-left)
            maxWater = max(maxWater,area)
            if height[left]<height[right]:
                left += 1
            else:
                right -= 1
        
        return maxWater

File: container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
     
        int maxArea = 0;
        int left = 0;
        int right = height.length - 1;
        int currArea = 0;
        
        while(left<right){
            currArea = Math.min(height[left],height[right])*(right-left);
            maxArea = Math.max(maxArea,currArea);
            
            if(height[left]<height[right]){
              left++;  
            }
            else{
                right--;
            }
        }
        
        return maxArea;
        
    }
}